### ISSUE TITLE
tests with #[should_panic] should print a message on failure

### ISSUE THREAD

**Issue by fbenkstein** [rust-lang/rust#60790]
When a `#[should_panic]` `#[test]` fails because there was no panic it should produce an error message to that effect.

For example, the following test:

```rust
#[test]
#[should_panic]
fn doesnt_actually_panic() {}
```

when run with `cargo test` produces the following output:

```
test doesnt_actually_panic ... FAILED

failures:

failures:
    doesnt_actually_panic
```

There is no indication why it failed.  Which can be problematic if the test itself is created by a macro.  Example: https://github.com/BurntSushi/fst/blob/master/src/raw/tests.rs#L131-L140

**Comment by varkor** [comment]
I'm not familiar with the code, but I'd guess that you just need to change `TrFailed` in here:
https://github.com/rust-lang/rust/blob/036e368d6883e0bc141c1b54578876cd5bfb2468/src/libtest/lib.rs#L1525-L1548
to `TrFailedMsg("test did not panic as expected")`
(And add a test to make sure it works.)

**Comment by chansuke** [comment]
Thanks.I would like to work on this.

**Comment by varkor** [comment]
@chansuke: go ahead! If you have any questions, feel free to ask them here.

**Comment by chansuke** [comment]
@varkor Hi.I've added the `TrFailedMsg("test did not panic as expected")` in `fn calc_result()` and trying to write a test but is this right approach?

```
    #[test]
    fn test_calc_result() {
        let args = // what should I write?
        let panic_message = "test did not panic as expected"
        let result = calc_result(args)
        assert_eq!(result, panic_message)
    }
```

**Comment by varkor** [comment]
@chansuke: actually, I think you can get away with just changing this line:
https://github.com/rust-lang/rust/blob/7158ed9cbea805adf8161d3deaadba2f85b7692e/src/libtest/lib.rs#L1925
to update it to your `TrFailedMsg`, because such a test already exists for this case.

**Comment by chansuke** [comment]
@varkor Thanks

**Comment by varkor** [comment]
The previous fix was closed, so this issue is open again. The [previous PR](https://github.com/rust-lang/rust/pull/61068) contains detailed instructions.

**Comment by kennethbgoodin** [comment]
I'd like to finish this out if @chansuke is no longer working out.
